block-producer/src/state_view/mod.rs

```diff
@@ -44,6 +45,8 @@ where
     S: Store,
 {
     // TODO: Verify proof as well
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip(self), ret, err(Debug), fields(COMPONENT))]
     async fn verify_tx(
         &self,
         candidate_tx: SharedProvenTx,
```

Comment:
Similar comments/questions to the ones above re needing to log `ret`, `err`, `COMPONENT` etc.